components/screens/browser-api/kaspa/SwitchKaspaNetwork.tsx

```diff
@@ -0,0 +1,72 @@
+import Splash from "@/components/screens/Splash";
+import { NetworkType } from "@/contexts/SettingsContext.tsx";
+import SwitchNetwork from "../SwitchNetwork";
+import { useSettings } from "@/hooks/useSettings";
+import { ApiExtensionUtils } from "@/api/extension";
+import { ApiUtils } from "@/api/background/utils";
+import { z } from "zod";
+
+export default function SwitchKaspaNetwork() {
+  const [settings, setSettings] = useSettings();
+
+  const requestId =
+    new URLSearchParams(window.location.search).get("requestId") ?? "";
+  const encodedPayload = new URLSearchParams(window.location.search).get(
+    "network",
+  );
+
+  const payload = encodedPayload ? decodeURIComponent(encodedPayload) : null;
+
+  const network = payload ? z.string().parse(payload) : null;
+
+  const loading = !requestId || !network;
+
+  const networks = [
```

Comment:
I feel like this array belongs to the `SwitchNetwork` component since you are only using the ID in this component and the rest is used in the child.

Reply:
I will split ethereum and kaspa networks first, it is for reusing the same UI.